skimage/morphology/misc.py

```diff
@@ -90,6 +94,10 @@ def remove_small_objects(ar, min_size=64, connectivity=1, *, out=None):
     out : ndarray, same shape and type as input `ar`
         The input array with small connected components removed.
 
+    See Also
+    --------
+    skimage.morphology.remove_near_objects
```

Comment:
I don't have a better name, so I'll just put this out there in case you have other ideas: but `remove near objects` isn't super intuitive, since we're not removing objects near to something else. The *types* of names that might capture it, but that doesn't fit nicely into the `remove_*_objects` mold, are:

- `reduce_object_density`
- `filter_objects` (could enclose both by size and by distance?)
- `sparsify_objects`

Reply:
Upon in-person discussion we settled for `remove_objects_by_distance`. It's not ideal but we feel like this makes it reasonably clear that the function deals with distances between objects instead of the property of a single one. 

The idea is to also rename `remove_small_objects` to `remove_objects_by_size` for a similar reason. 1) "small" isn't really a good term to use in this context; what is small? And 2) to keep things consistent.